src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -119,52 +126,33 @@ void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
     var_types      targetType  = node->TypeGet();
     var_types      baseType    = node->gtSIMDBaseType;
 
-    regNumber op1Reg = op1->gtRegNum;
-    regNumber op2Reg = REG_NA;
-    emitter*  emit   = getEmitter();
+    instruction ins    = INS_invalid;
+    regNumber   op1Reg = op1->gtRegNum;
+    regNumber   op2Reg = REG_NA;
+    emitter*    emit   = getEmitter();
 
     genConsumeOperands(node);
 
     switch (intrinsicID)
     {
         case NI_SSE2_Add:
-        {
-            op2Reg = op2->gtRegNum;
+            assert((baseType == TYP_BYTE) || (baseType == TYP_UBYTE) ||
```

Comment:
This check should be a side effect of `insOfHWIntrinsic`. If the type is not supported (e.g. `TYP_REF`) `insOfHWIntrinsic` should assert. If a specific intrinsic does not support a specific type then the corresponding instruction should be `INS_invalid` and that too should assert.

Reply:
Will change.

Reply:
Fixed.